app/assets/javascripts/components/subjects_taught.js

```diff
@@ -0,0 +1,40 @@
+"use strict";
+
+document.addEventListener("DOMContentLoaded", function() {
+  var fieldset = document.querySelector("#tslr_claim_eligible_subjects");
+  var el;
+  var eligibleSubjectsSelector;
+  var notTeachingSubjectsSelector;
+
+  if (!fieldset) {
+    return;
+  }
+
+  fieldset.addEventListener(
+    "click",
+    function(e) {
+      el = e.target;
+
+      if (!el) {
+        return;
+      }
+      if (el.checked === false) {
+        return;
+      }
+
+      eligibleSubjectsSelector = "tslr_claim[eligible_subjects][]";
+      notTeachingSubjectsSelector = "tslr_claim[mostly_teaching_eligible_subjects]";
```

Comment:
These are specifically `name`s not selectors, so let's name the variables to match that.